tests/dataset_builders/pie/test_lisa.py

```diff
@@ -0,0 +1,368 @@
+from pie_datasets.builders.brat import BratDocument, BratDocumentWithMergedSpans
+from pie_modules.document.processing import tokenize_document
+from pytorch_ie.annotations import BinaryRelation, LabeledMultiSpan, LabeledSpan
+from pytorch_ie.core import AnnotationList, annotation_field
+from pytorch_ie.documents import TextDocumentWithLabeledSpansAndBinaryRelations, TokenBasedDocument
+from transformers import AutoTokenizer, PreTrainedTokenizer
+
+from dataset_builders.pie.lisa.lisa import Lisa
+from tests.dataset_builders import PIE_BASE_PATH
+
+
+@dataclasses.dataclass
+class TestTokenDocumentWithLabeledSpans(TokenBasedDocument):
+    labeled_spans: AnnotationList[LabeledSpan] = annotation_field(target="tokens")
+
+
+@dataclasses.dataclass
+class TestTokenDocumentWithLabeledSpansAndBinaryRelations(TestTokenDocumentWithLabeledSpans):
+    binary_relations: AnnotationList[BinaryRelation] = annotation_field(target="labeled_spans")
+
+
+disable_caching()
+
+DATASET_NAME = "lisa"
+PIE_DATASET_PATH = PIE_BASE_PATH / DATASET_NAME
+SPLIT_SIZES = {"train": 118}
+
+
+@pytest.fixture(scope="module", params=["default", "merge_fragmented_spans"])
+def dataset_variant(request) -> str:
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def dataset(dataset_variant) -> DatasetDict:
+    return DatasetDict.load_dataset(str(PIE_DATASET_PATH), name=dataset_variant)
+
+
+def test_dataset(dataset):
+    assert dataset is not None
+    assert {name: len(ds) for name, ds in dataset.items()} == SPLIT_SIZES
+
+
+@pytest.fixture(scope="module")
+def document(dataset) -> Union[BratDocument, BratDocumentWithMergedSpans]:
+    return dataset["train"][0]
+
+
+def test_document(document, dataset_variant):
+    assert document is not None
+    assert document.text.startswith(
+        "Hallo liebe <user>, danke für deine lieben Worte, hier im Forum sind alle so nett und hilfsbereit."
+    )
+    if dataset_variant == "default":
+        assert isinstance(document, BratDocument)
+        span = document.spans[0]
+        assert isinstance(span, LabeledMultiSpan)
+        # TODO
+    elif dataset_variant == "merge_fragmented_spans":
+        assert isinstance(document, BratDocumentWithMergedSpans)
+        span = document.spans[0]
+        assert isinstance(span, LabeledSpan)
+        # TODO
+    else:
+        raise ValueError(f"Unknown dataset variant: {dataset_variant}")
+
+
+def my_converter(document: BratDocument) -> TextDocumentWithLabeledSpansAndBinaryRelations:
+    spans = document.spans
+    new_doc = TextDocumentWithLabeledSpansAndBinaryRelations(text=document.text)
+    new_spans = []
+    old2new_span = defaultdict(list)
+    for span in spans:
+        slices = span.slices
+        label = span.label
+        score = span.score
+        for slice in slices:
+            new_span = LabeledSpan(start=slice[0], end=slice[1], label=label, score=score)
+            new_spans.append(new_span)
+            old2new_span[span].append(new_span)
```

Comment:
you should also add new relations, e.g. with label `parts_of_same`, that connect all the fragments. I would recommend to do this strategy: just connect the neighboring fragments, so this does not create too many new relations

Reply:
so for instance, two fragments of an entity having a relation with another two-fragmented entity will result in three new relations—two `parts_of_same` between fragments of respective entities and one between the last fragment of the first entity and the first fragment of the second entity.

Reply:
hmm my gut feeling says, that I would still create all combinations of head and tail with the original relation label (as you have written it). Just for the newly introduced `parts_of_same` relations I would create only neighboring ones. Some background: the relation extraction taskmodule creates dummy relations with label `no_relation` for all combinations of entities that are not already in a relation. My hypothesis is that a model can "easily" learn to predict the `parts_of_same` relation only for neighboring parts, but it is hard to learn that a relation holds only between the first fragment of the head and the first fragment of the tail, but not between the other parts. But this is primarily a gut feeling, I do not have experimental evidence.

Reply:
Thanks for the background, then I'll just add `parts_of_same` in neighboring fragments in addition to already created new relation combinations.